Repo/TodoRepo.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.AspNetCore.Mvc;
+using todoDotNet6.Repo;
+
+namespace todoDotNet6.TodoRepo
+{
+
+    public interface ITodoRepository : IRepository<Todo,Guid> {}
+
+    public class TodoRepo : ControllerBase, ITodoRepository
+    {
+        public static Dictionary<Guid, Todo> todos = new Dictionary<Guid, Todo>() {};
+
+        public Todo CreateTodo(Todo request)
+        {
+            request.Id = Guid.NewGuid();
+            request.CreatedDate = DateTime.Now;
+            todos.Add(request.Id, request);
+            return request;  
+        }
+
+        public Dictionary<Guid, Todo> DeleteTodo(Guid id)
+        {
+            todos.Remove(id);
+            return todos;
+        }
+
+        public Todo GetATodo(Guid id)
+        {
+            if (!todos.ContainsKey(id))
+            {
+                return null;
+            }
+
+            return todos[id];
+        }
+
+        public Dictionary<Guid, Todo> GetTodos()
```

Comment:
See my comment above as well. Here the calling code does need all TODOs because thats the point of `GetTodos` but all it needs is a list of TODOs. Usage of the dictionary is an **internal implementation detail** of your repository. No other class should know this